linode/manager
# refactor: [M3-8002] - Remove lodash (global import) as a package dependency

## Description 📝
We are currently importing the whole lodash library for a few utils
- lodash is large, and there's no reason to load the whole bundle for a few utils
- some of those utils being quite simple to write ourselves
- the more complicated ones can be imported indivually

## Changes  🔄
- removed lodash as a global dependency
- import complex utils from individual lodash packages
- backfill utils easily convertible from lodash
  - import replacements in components
  - write test for each

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![_private_var_folders_16_z582qxlx7bz20rfzy69bzll00000gp_T_tmp-28688-o00QO6UrXUBk_stats html](https://github.com/linode/manager/assets/130582365/9c0ee1e0-5f80-4962-bc34-e880b64601d0) | ![_private_var_folders_16_z582qxlx7bz20rfzy69bzll00000gp_T_tmp-27768-1RrNV9piBIiO_stats html](https://github.com/linode/manager/assets/130582365/8ef58254-ca33-4e0a-a5d3-e5a6f12200b1) |
| index.js bundle size:  **1906.61** kB  | index.js bundle size:  **1,825.27** kB |

## How to test 🧪
- ℹ️  I ran deep compares for every util replaced, making sure the data returned was identical. I also tested the new unit tests with both the lodash and the new util
- ℹ️  This refactor assumes no regression (functional or visual) as a result

### Verification steps
- pull code and run `yarn && yarn up`

I frankly trust our suite, and foresee no regression however we can verify the areas where the lodash utils have been replaced:
- Secret Token Dialog (CopyableAndDownloadableTextField.tsx)
- Display Price (Currency.tsx)
- Link
- Table
- Backup Drawer
- Add Node Pool & Resize Node Pool Drawers
- Linode Create flow (from stackscript)
- Linode Detail (multiple public IPs)

## As an Author I have considered 🤔

*Check all that apply*
